patches/minecraft/net/minecraft/client/resources/model/WeightedBakedModel.java.patch

```diff
@@ -53,6 +53,13 @@
 +
 +   public BakedModel applyTransform(net.minecraft.client.renderer.block.model.ItemTransforms.TransformType transformType, com.mojang.blaze3d.vertex.PoseStack poseStack, boolean applyLeftHandTransform) {
 +      return this.f_119542_.applyTransform(transformType, poseStack, applyLeftHandTransform);
++   }
++
++   @Override // FORGE: Get render types based on the active weighted model
++   public net.minecraftforge.client.ChunkRenderTypeSet getRenderTypes(@org.jetbrains.annotations.NotNull BlockState state, @org.jetbrains.annotations.NotNull RandomSource rand, @org.jetbrains.annotations.NotNull net.minecraftforge.client.model.data.ModelData data) {
++      return WeightedRandom.m_146314_(this.f_119541_, Math.abs((int)rand.m_188505_()) % this.f_119540_)
```

Comment:
Would there be problems if the baked model used by `getQuads` differs from the baked model used in this method? I'm thinking of the situation where something calls `getQuads` with a random source, and calls `getRenderTypes` with the same random source afterwards, meaning that both methods have a different random number and therefore can select a different baked model from each other.

Reply:
While a fair concern, it would be a breach of the standard model contract, where the seed is expected to be reset to a known value before each method call, so it's a non-problem.